pkgs/build-support/ocaml/dune.nix

```diff
@@ -40,6 +40,10 @@ stdenv.mkDerivation ({
 
   nativeBuildInputs = [ ocaml Dune findlib ] ++ nativeBuildInputs;
 
+  buildInputs = buildInputs ++ (if doCheck then checkInputs else []);
```

Comment:
If you drop this, what's the fallout? I'm also not keen on keeping this tbh

Reply:
it breaks everything that uses a unit test library (ounit, alcotest...). I agree that such a hack is undesirable, so in my mind this is blocked on https://github.com/NixOS/nixpkgs/issues/161570